### DISCUSSION TITLE
Looking forward to contribute

### DISCUSSION THREAD

**Discussion by akourieh** [ImageMagick/ImageMagick#7772] [General]
I am a computer science student and I would like to contribute to this project for a university course about Open-source development. I would like something easy to start with like documenting (readme, ...) or fixing some easy issues (shell, Python, C/C++, Java). Thank you for your help.

**Comment by snibgo** [comment]
Good stuff. Have you looked at https://github.com/ImageMagick/ImageMagick/discussions?discussions_q=is%3Aopen+label%3A%22up+for+grabs%22 ?

**Reply by akourieh** [reply]
Yes I have, I thought the ones I found interesting for me were already dealt with or didn't understand them. For example, [this one #7026](https://github.com/ImageMagick/ImageMagick/discussions/7026#discussion-6069262) seems to ask for silencing warnings but I am not sure where and if still needed. There is this one person who required to change the doc [here #6673](https://github.com/ImageMagick/ImageMagick/discussions/6673#discussioncomment-7080918) but I don't know if this is actually needed also. And finally the [#6704](https://github.com/ImageMagick/ImageMagick/discussions/6704#discussion-5661010) I think requires to add docstring to the C++ functions which I am not sure if I am able to write them correctly.

**Reply by dlemstra** [reply]
The first example that you mention won't be that easy to fix. The moments documentation has been implemented by printing a warning on the console. I just removed the `up for grabs` label there.

Feel free to give 6704 a try we can help you with this. And maybe you could also add something to the GitHub pipeline that checks if the docstring comments are written in the correct syntax?

**Reply by akourieh** [reply]
Sorry for the late reply, yes I would like to help for the #6704. However, I have never heard of a Github pipeline, is this what you mean : https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions ? If so, I would have to make a new .yml in the .github/workflows to check the docstring of the code in Magick++/ with a script, right ?

**Reply by dlemstra** [reply]
No need to apologize for the late reply I will also not respond right away. If you never heard of a GitHub pipeline you might want to read or watch some videos about it. And it would probably be a good idea to include those checks in an existing build. My advice would be to fork our repository and play around with it. You can always make a new fork of the repository.

**Reply by akourieh** [reply]
Hello, I have checked the code and I saw that the HTML doc files (like Image++.html) were made statically and does not refer to the function signatures of the headers. So, should I make a script to check among all functions in the www/*.html files and match their signature to the ones in the headers (in Magick++) which seems quite complexe for me to be honest. Or, should I make a pipeline to generate docstring with doxygen like in the Makefile with "make html-local" after a commit gets pushed in the main.

**Reply by dlemstra** [reply]
I don't think we want to do this every commit automatically. Maybe we could have something that could create/update our docs and report an error in the pipeline when we need to update the docs?
